src/Sylius/Bundle/ApiBundle/Form/Extension/ApiProductTypeExtension.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ApiBundle\Form\Extension;
+
+use Sylius\Bundle\ApiBundle\Form\EventSubscriber\RemoveVariantSelectionFieldFormSubscriber;
+use Sylius\Bundle\ProductBundle\Form\Type\ProductType;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\FormBuilderInterface;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class ApiProductTypeExtension extends AbstractTypeExtension
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder->addEventSubscriber(new RemoveVariantSelectionFieldFormSubscriber());
```

Comment:
To be honest, we could remove `RemoveVariantSelectionFieldFormSubscriber` class and use callable instead:
```php
->addEventListener(FormEvents::POST_SUBMIT, function (FormEvent $event) {
    $data = $event->getData();

    if (!array_key_exists('variantSelectionMethod', $data)) {
        $form = $event->getForm();
        $form->remove('variantSelectionMethod');
    }
})
```